torch/csrc/cuda/nccl.h

```diff
@@ -6,55 +6,87 @@
 #include <c10/cuda/CUDACachingAllocator.h>
 #include <c10/util/Optional.h>
 
-#include <nccl.h>
-
 #include <cstddef>
 #include <vector>
 
+typedef struct ncclComm* ncclComm_t;
+
 namespace torch {
 namespace cuda {
 namespace nccl {
 
+    /*
+The below enums are from
+https://github.com/NVIDIA/nccl/blob/master/src/nccl.h.in
+however they're declare anyonomously there, so we can't forward declare them
+here.*/
+/* Error type */
+typedef enum {
```

Comment:
Nit: it would be nice to have those as properly namespace scoped `enum classes` rather than global C style enums

Reply:
fixed